source/gameengine/GameLogic/SCA_JoystickSensor.cpp

```diff
@@ -374,48 +352,42 @@ PyObject *SCA_JoystickSensor::pyattr_get_axis_single(void *self_v, const KX_PYAT
 	return PyLong_FromLong(joy ? joy->GetAxisPosition(self->m_axis - 1) : 0);
 }
 
+PyObject *SCA_JoystickSensor::pyattr_check_hat(void *self_v, const KX_PYATTRIBUTE_DEF *attrdef)
+{
+	ShowDeprecationWarning("SCA_JoystickSensor.hat", "SCA_JoystickSensor.button");
+	return NULL;
```

Comment:
I don't know about throwing an error for compatibility issues. Can you do a kind of conversion from buttons ? if not i prefer to return a dumb value like 0 every times instead of raise an error.